src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -165,23 +302,28 @@ static u8 sAnimCloseFrames[2 * PLAYER_FORM_MAX] = {
     60, // PLAYER_FORM_DEKU
     70, // PLAYER_FORM_HUMAN
 };
-
-static Gfx* D_808679A4[14][2] = {
-    { gDoorLeftDL, gDoorRightDL },
-    { gWoodfallDoorDL, gWoodfallDoorDL },
-    { gObservatoryLabDoorDL, gObservatoryLabDoorDL },
-    { gZoraHallDoorDL, gZoraHallDoorDL },
-    { gSwampDoorDL, gSwampDoorDL },
-    { gMagicHagPotionShopDoorDL, gMagicHagPotionShopDoorDL },
-    { gLotteryCuriosityShopMayorHouseDoorDL, gLotteryCuriosityShopMayorHouseDoorDL },
-    { gPostOfficeTradingPostDoorDL, gPostOfficeTradingPostDoorDL },
-    { gInnSchoolDoorDL, gInnSchoolDoorDL },
-    { gMilkBarDoorDL, gMilkBarDoorDL },
-    { gMusicBoxHouseDoorDL, gMusicBoxHouseDoorDL },
-    { gPiratesFortressDoorDL, gPiratesFortressDoorDL },
-    { gOceansideSpiderHouseDoorDL, gOceansideSpiderHouseDoorDL },
-    { gFieldWoodDoorLeftDL, gFieldWoodDoorRightDL },
+static_assert(ARRAY_COUNT(sAnimCloseFrames) == 2 * PLAYER_FORM_MAX,
+              "The entry count of `sAnimCloseFrames` should be exactly twice as PLAYER_FORM_MAX");
+
+static Gfx* sDoorDLists[][2] = {
+    { gDoorLeftDL, gDoorRightDL },                            // DOOR_DL_DEFAULT
+    { gWoodfallDoorDL, gWoodfallDoorDL },                     // DOOR_DL_WOODFALL
+    { gObservatoryLabDoorDL, gObservatoryLabDoorDL },         // DOOR_DL_OBSERVATORY_LAB
+    { gZoraHallDoorDL, gZoraHallDoorDL },                     // DOOR_DL_ZORA_HALL
+    { gSwampDoorDL, gSwampDoorDL },                           // DOOR_DL_SWAMP
+    { gMagicHagPotionShopDoorDL, gMagicHagPotionShopDoorDL }, // DOOR_DL_MAGIC_HAG_POTION_SHOP
+    { gLotteryCuriosityShopMayorHouseDoorDL,
+      gLotteryCuriosityShopMayorHouseDoorDL },                      // DOOR_DL_LOTTERY_CURIOSITY_SHIP_MAYOR_HOUSE
+    { gPostOfficeTradingPostDoorDL, gPostOfficeTradingPostDoorDL }, // DOOR_DL_POST_OFFICE_TRAIDING_POST
+    { gInnSchoolDoorDL, gInnSchoolDoorDL },                         // DOOR_DL_INN_SCHOOL
+    { gMilkBarDoorDL, gMilkBarDoorDL },                             // DOOR_DL_MILK_BAR
+    { gMusicBoxHouseDoorDL, gMusicBoxHouseDoorDL },                 // DOOR_DL_MUSIC_BOX
+    { gPiratesFortressDoorDL, gPiratesFortressDoorDL },             // DOOR_DL_PIRATES_FORTESS
+    { gOceansideSpiderHouseDoorDL, gOceansideSpiderHouseDoorDL },   // DOOR_DL_OCEANSIDE_SPIDER_HOUSE
+    { gFieldWoodDoorLeftDL, gFieldWoodDoorRightDL },                // DOOR_DL_DEFAULT_FIELD_KEEP
 };
+static_assert(ARRAY_COUNT(sDoorDLists) == DOOR_DL_MAX,
+              "The entry count of `sDoorDLists` should match the `EnDoorDListIndex` enum");
```

Comment:
Instead of using the `DOOR_DL_MAX` as the size of the array I'm using an `static_assert` to make sure the size matches what it should, preventing silent issues